### PR TITLE
Removed throwing an exception when the installation process is closed…

### PR DESCRIPTION
**Pull request by genI3**
… or canceled by user at language selection stage.

### CODE REVIEW

File: dotNetInstaller/ConfigFileManager.cpp [hunk 1]
```diff
@@ -59,25 +59,33 @@ int ConfigFileManager::Run()
...
-        case IDCANCEL:
-            THROW_EX(L"Language selection cancelled by user");
-        }
+		if (lsdlg.DoModal() == IDOK)
+			return LanguageSelection_Selected;
+		else
+			// Close or cancel of the setup process at the language selection 
+			// stage is a normal situation and it does not throw error!
+			return LanguageSelection_Cancel;
+		//else
```
Comment (dblock) [inline dotNetInstaller/ConfigFileManager.cpp]: Delete commented out code please.

**Comment by dblock** [comment]
This looks good. It needs an entry in https://github.com/dblock/dotnetinstaller/blob/master/CHANGELOG.md please, and possibly documentation changes. I also believe we have tests around this, does everything pass? Maybe add a test for the cancelation story if it's not too difficult too.

**Comment by genI3** [comment]
I will delete the commented out code and add a test for the cancellation at next commit.
In what format do I make changes to the changelog?

**Comment by dblock** [comment]
Just make it look like every other line in there, thanks.

**Comment by genI3** [comment]
Done.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ TBD
 Features
 --------
 
+* [#112](https://github.com/dblock/dotnetinstaller/pull/112) - Removed throwing an exception when the installation process is closed or canceled by user at language selection stage. Added unit test for testing canceling the setup at language selection stage. - [@genI3](https://github.com/genI3).
```
Comment (dblock) [inline CHANGELOG.md]: Remove period after "stage"
Remove "Added unit test...".

File: UnitTests/dotNetInstallerLibUnitTests/ConfigFilesImpl.h [hunk 3]
```diff
@@ -20,7 +21,8 @@ namespace DVLib
 			bool OnRunConfiguration(const ConfigurationPtr& configuration);
 			int GetRuns() const { return m_runs; }
 			int GetDownloads() const { return m_downloads; }
-			bool OnSelectLanguage() { return true; }
+			void SetLanguageSelectionCancelled() { m_langselection = LanguageSelection_Cancel;}
```
Comment (dblock) [inline UnitTests/dotNetInstallerLibUnitTests/ConfigFilesImpl.h]: Missing space after `;`

File: UnitTests/dotNetInstallerLibUnitTests/ConfigFilesUnitTests.cpp [hunk 4]
```diff
@@ -110,6 +110,19 @@ void ConfigFilesUnitTests::testSelectLanguageNot1040()
     CPPUNIT_ASSERT(cf[0]->lcid_filter == L"!1040");
 }
 
+void ConfigFilesUnitTests::testSelectLanguageCancelled()
```
Comment (dblock) [inline UnitTests/dotNetInstallerLibUnitTests/ConfigFilesUnitTests.cpp]: Something wrong with tabs/spaces here, it's a nitpick but I'd like to keep things consistent, please.

**Comment by dblock** [comment]
Do I understand correctly that this only affects a UI-full installation?

Do we need any changes in https://github.com/dblock/dotnetinstaller/blob/master/Documentation/Content/Multilingual.aml?

I also made two nitpicky comments, sorry for being a hassle. I'd appreciate if you could fix space/tabs here so we don't start having a codebase that looks like it's all over the place.

**Comment by dblock** [comment]
@icnocop You're still using DNI? Want to take a look/merge?

**Comment by genI3** [comment]
Now everything is OK?

**Comment by genI3** [comment]
> Do I understand correctly that this only affects a UI-full installation? Before a user would see an error, now they just see setup canceled, right?

Only in the case when we see the language selection window. Right.

> Do we need any changes in https://github.com/dblock/dotnetinstaller/blob/master/Documentation/Content/Multilingual.aml?

On this occasion, I do not know anything.

**Comment by dblock** [comment]
Looks good, merging.

@genI3 Are you actively using this? I haven't in years, we could use some help co-maintaining the project, maybe making a release?

**Comment by genI3** [comment]
I use it from time to time to build the bootstrapper of the installer. Regarding the release of the dotNetInstaller - I don't know.

**Pull request merged**